h/static/scripts/directive/annotation-share-dialog.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var angular = require('angular');
+
+module.exports = function () {
+  return {
+    bindToController: true,
+    controllerAs: 'vm',
+    restrict: 'E',
+    template: require('../../../templates/client/annotation_share_dialog.html'),
+    controller: ['$timeout', '$scope', function ($timeout, $scope) {
+      this.copyToClipboard = function (event) {
+        var $container = angular.element(event.currentTarget).parent();
+        var shareLinkInput = $container.find('input')[0];
+
+        try {
+          shareLinkInput.select();
+          document.execCommand('copy');
+
+          $scope.copyToClipboardMessage = 'Link copied to clipboard!';
```

Comment:
Rather than setting these properties on the scope directly, it would be slightly more idiomatic to set them on the controller, then access them in the scope via `vm.copyToClipboardMessage` for example.